wire-ios/Wire-iOS/Sources/UserInterface/StartUI/Common/EmptySearchResultsView.swift

```diff
@@ -43,46 +47,83 @@ extension EmptySearchResultsViewAction {
     }
 }
 
+// MARK: - EmptySearchResultsViewDelegate
+
 protocol EmptySearchResultsViewDelegate: AnyObject {
     func execute(action: EmptySearchResultsViewAction, from: EmptySearchResultsView)
 }
 
+// MARK: - EmptySearchResultsView
+
 final class EmptySearchResultsView: UIView {
 
     typealias LabelColors = SemanticColors.Label
 
-    private var state: EmptySearchResultsViewState = .noUsers {
+    // MARK: - Computed Properties
+
+    private var state: EmptySearchResultsViewState = .initialSearch {
         didSet {
-            iconView.image = icon
-            iconView.tintColor = iconColor
-            statusLabel.text = self.text
+            updateUIForCurrentEmptySearchResultState()
+        }
+    }
+
+    private var text: String {
+        typealias Message = L10n.Localizable.Peoplepicker.NoMatchingResults.Message
 
-            if let action = self.buttonAction {
-                actionButton.isHidden = false
-                actionButton.setup(title: action.title)
+        switch state {
+        case .everyoneAdded:
+            return Message.usersAllAdded
+        case .noUsers:
+            if isFederationEnabled {
+                return Message.usersAndFederation
+            } else {
+                return Message.users
+            }
+        case .noServices:
+            return Message.services
+        case .noServicesEnabled:
+            if isSelfUserAdmin {
+                return Message.servicesNotEnabledAdmin
             } else {
-                actionButton.isHidden = true
+                return Message.servicesNotEnabled
             }
+        case .initialSearch:
+            return ""
         }
     }
 
-    func updateStatus(searchingForServices: Bool, hasFilter: Bool) {
-        switch (searchingForServices, hasFilter) {
-        case (true, false):
-            self.state = .noServicesEnabled
-        case (true, true):
-            self.state = .noServices
-        case (false, true):
-            self.state = .noUsers
-        case (false, false):
-            self.state = .everyoneAdded
+    private var icon: UIImage {
+        let icon: StyleKitIcon
+
+        switch state {
+        case .initialSearch:
+            return UIImage()
+        case .noServices, .noServicesEnabled:
+            icon = .bot
+        default:
+            icon = .personalProfile
+        }
+
+        return icon.makeImage(size: .large, color: iconColor)
+    }
+
+    private var buttonAction: EmptySearchResultsViewAction? {
+        switch state {
+        case .noServicesEnabled where isSelfUserAdmin:
+            return .openManageServices
+        case .noUsers:
+            return .openSearchSupportPage
+        default:
+            return nil
         }
     }
 
+    // MARK: - Properties
+
     private let isSelfUserAdmin: Bool
     private let isFederationEnabled: Bool
 
-    /// Contains the `stackView`.
+    // Contains the `stackView`.
```

Comment:
fyi (non-blockin) : if you remove the 3rd slash I don't think it gets shown in the help description of Xcode

Reply:
Fixed!